enhancements/olm-admission-webhooks.md

```diff
@@ -0,0 +1,135 @@
+
+### CSV support for Validating and Mutating Webhook
+
+CSV support for webhooks will begin with the introduction of the `OLMWebhook` and `WebhookDescription` structs. These structures provides OLM with sufficient information to create either mutating or validating admission webhooks. These structs should look similar to the following examples:
+
+```go
+// OLMWebhook is an exact replica of the MutatingWebhook struct with the excpetion that the WebhookClientConfig is missing, as OLM will generate this resource itself.
+// This struct MUST contain all fields that are present in the ValidataingWebhook and MutatingWebhook structs. 
+type OLMWebhook struct {
+	Name                    string                                          `json:"name"`
+	Rules                   []admissionregistrationv1.RuleWithOperations    `json:"rules"`
+	FailurePolicy           *admissionregistrationv1.FailurePolicyType      `json:"failurePolicy,omitempty"`
+	MatchPolicy             *admissionregistrationv1.MatchPolicyType        `json:"matchPolicy,omitempty"`
+	NamespaceSelector       *metav1.LabelSelector                           `json:"namespaceSelector,omitempty"`
+	ObjectSelector          *metav1.LabelSelector                           `json:"objectSelector,omitempty"`
+	SideEffects             *admissionregistrationv1.SideEffectClass        `json:"sideEffects"`
+	TimeoutSeconds          *int32                                          `json:"timeoutSeconds,omitempty"`
+	AdmissionReviewVersions []string                                        `json:"admissionReviewVersions"`
+	ReinvocationPolicy      *admissionregistrationv1.ReinvocationPolicyType `json:"reinvocationPolicy,omitempty"`
+}
+
+// WebhookDescription is used to provide OLM with information about the Webhook being shipped with the operator
+type WebhookDescription struct {
+    // Type is used to identify the admission webhook as a validating or mutating webhook. Accepted values include `validating` and `mutating`
+    Type                    WebhookAdmissionType                            `json:"type"` #TODO: Should this be in the OLMWebhook struct???
+    // DeploymentName must match one of the deployments included in the CSV.
+	DeploymentName          string                                          `json:"deploymentName,omitempty"`
+    // ContainerPort specifies which port the service should forward requests to.
+	ContainerPort           int32                                           `json:"containerPort,omitempty"`
+    // webhook contains the information that must be sent to the validating or mutating webhook. 
+	webhook                 OLMWebhook                                      `json:"webhook"`
```

Comment:
`Webhook` should be capitalized

Reply:
Good point - addressed in the latest version of the PR.